### ISSUE TITLE
Add an edge detection function to the suite of ImageEffects

### ISSUE THREAD

**Issue by andrewyancey** [andrewyancey/ImageManipulation#6]
Edge detection works by detecting an abrupt change in pixel colors. This would be a nice addition to the suite of image effects

**Comment by oneofmanylukes** [comment]
We could implement it as stated in this [this](https://www.reddit.com/r/csharp/comments/ceo3fq/im_working_on_a_image_effects_application_for/eu6meox/) reddit comment, then we could use the implementation in many different ways.

**Comment by genious97** [comment]
Kappa this was exactly what I mentioned in the discord

**Comment by andrewyancey** [comment]
Yes. It’s just a matter of when someone gets around to implement it.
